govcd/edgegateway.go

```diff
@@ -1199,7 +1278,40 @@ func (egw *EdgeGateway) GetVnicIndexByNetworkNameAndType(networkName, networkTyp
 	if err != nil {
 		return nil, fmt.Errorf("cannot retrieve vNic configuration: %s", err)
 	}
-	return GetVnicIndexByNetworkNameAndType(networkName, networkType, vnics)
+	return getVnicIndexByNetworkNameAndType(networkName, networkType, vnics)
+}
+
+// GetAnyVnicIndexByNetworkName returns *int of vNic index and network type by network name
```

Comment:
I would like to see explanation of vNic is and how it is mapping to edge gtw context.

Reply:
This could be a bit lengthy instruction. In fact the unit test has a sample XML config how it looks like. https://github.com/vmware/go-vcloud-director/pull/247/files#diff-5ae225082b593473c248452f5c3b8b2cR297

I have adjusted the comment to just touch the point that XML contains a mapping of vNic (network interfaces on edge gateway) and network names and their types.